homeassistant/components/tellstick.py

```diff
@@ -41,101 +41,110 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
-def _discover(hass, config, found_devices, component_name):
+def _discover(hass, config, component_name, found_tellcore_devices):
     """Setup and send the discovery event."""
-    if not len(found_devices):
+    if not len(found_tellcore_devices):
         return
 
-    _LOGGER.info(
-        "Discovered %d new %s devices", len(found_devices), component_name)
+    _LOGGER.info("Discovered %d new %s devices", len(found_tellcore_devices),
+                 component_name)
 
     signal_repetitions = config[DOMAIN].get(ATTR_SIGNAL_REPETITIONS)
 
     discovery.load_platform(hass, component_name, DOMAIN, {
-        ATTR_DISCOVER_DEVICES: found_devices,
+        ATTR_DISCOVER_DEVICES: found_tellcore_devices,
         ATTR_DISCOVER_CONFIG: signal_repetitions}, config)
 
 
 def setup(hass, config):
     """Setup the Tellstick component."""
-    # pylint: disable=global-statement, import-error
-    global TELLCORE_REGISTRY
-
-    import tellcore.telldus as telldus
-    import tellcore.constants as tellcore_constants
+    from tellcore.constants import TELLSTICK_DIM
     from tellcore.library import DirectCallbackDispatcher
+    from tellcore.telldus import TelldusCore
 
-    core = telldus.TelldusCore(callback_dispatcher=DirectCallbackDispatcher())
+    global TELLCORE_REGISTRY
```

Comment:
We have moved away from using global variables. Instead you can use the dict at `hass.data` for this.

```python
hass.data['tellcore_registry'] = …
```